scalameter-core/src/main/scala/org/scalameter/picklers/Pickler.scala

```diff
@@ -0,0 +1,18 @@
+package org.scalameter.picklers
+
+import scala.annotation.implicitNotFound
+
+
+@implicitNotFound(msg ="""No Pickler available for ${T}. Please define yours or
+import org.scalameter.picklers.nopickler._ and use SerializationPersistor.""")
+abstract class Pickler[T] {
+  def pickle(x: T): Array[Byte]
+
+  def tryUnpickle(a: Array[Byte]): (T, Array[Byte])
```

Comment:
I think that, in hindsight, it might have been better to have `unpickle(a: Array[Byte], from: Int, len: Int)`. This would allow more efficient `unpickle` for serializers that work directly on byte arrays, as you wouldn't have to allocate all these intermediate arrays.

Thoughts on this?